sdk/python/kfp/components/_dsl_bridge.py

```diff
@@ -22,8 +22,8 @@ def create_container_op_from_task(task_spec: TaskSpec):
     argument_values = task_spec.arguments
     component_spec = task_spec.component_ref._component_spec
 
-    if hasattr(component_spec.implementation, 'graph'):
-        raise TypeError('Cannot convert graph component to ContainerOp')
+    if not hasattr(component_spec.implementation, 'container'):
```

Comment:
why not checking if it is ContainerImplementation type?

Reply:
AFAIK, this is bit more pythonic. Normally, python programs should not do type checking at all, but just access the attributes (which will raise error if the object is incompatible). In this case however I want to show a human readable error message instead of the standard "object does not have attribute container".

I can change this to type checking if you want though.

Reply:
Done.